tests/ui/manual_unwrap_or.rs

```diff
@@ -80,4 +80,61 @@ fn unwrap_or() {
     };
 }
 
+fn result_unwrap_or() {
```

Comment:
Could we add some cases where the scrutinee is a binding?

Reply:
I added two `Err(s) => ` cases. Is this what you meant?

Reply:
Sorry for not being precise, I was refering to something like:
```rust
let a = Ok::<i32, &str>(1);
match a {
// ...
```
But I think the cases you added make sense too, let's keep them.

Reply:
Great remark, the suggestion unnecessarily wraps the binding in parens: `(a).unwrap_or(42)`. I'll change that.